File: forex.insights.spa/src/api/forex-alert/interfaces/forex-alert-search-response.ts
import { ForexAlertGetResponse } from "./forex-alert-get-response";

/**
 * Forex alert search response.
 */
interface ForexAlertSearchResponse {
    /**
     * Requested forex alerts.
     */
    forexAlerts: ForexAlertGetResponse[];
}

export type { ForexAlertSearchResponse };

File: forex.insights.api/DataModels/ForexAlerts/ForexAlertSearchResponse.cs
using forex.insights.api.Entities.ForexAlerts;

namespace forex.insights.api.DataModels.ForexAlerts
{
    /// <summary>
    /// Forex Alert search response.
    /// </summary>
    public class ForexAlertSearchResponse(IEnumerable<ForexAlert> forexAlerts)
    {
        /// <summary>
        /// Requested Forex Alerts.
        /// </summary>
        public IEnumerable<ForexAlertGetResponse> ForexAlerts { get; set; } =
            forexAlerts.Select(forexAlert => new ForexAlertGetResponse(forexAlert));
    }
}
